### PR TITLE
Add /metrics with a couple of run metrics

### PR DESCRIPTION
**Pull request by j16r**
Added the following metrics:

- head_tracker_heads_received
- job_subscriber_subscriptions
- run_manager_runs_started
- run_manager_runs_resumed
- run_manager_runs_cancelled
- run_queue_runs_queued
- run_queue_queue_size
- stats_pusher_events_sent
 
as well as a host given to us by the middleware, such as go threads, request size histogram, requests served etc.

### CODE REVIEW

File: core/services/run_manager.go [hunk 1]
```diff
@@ -357,6 +376,7 @@ func (jm *runManager) updateAndTrigger(run *models.JobRun) error {
 		return err
 	}
 	if run.Status == models.RunStatusInProgress {
+		runsResumed.Inc()
```
Comment (dimroc) [inline core/services/run_manager.go]: Curious if `runsResumed` will include scenarios like `pending_confirmation`, `pending_sleep`, etc

Comment (j16r) [reply core/services/run_manager.go]: All of the above.

File: core/services/head_tracker.go [hunk 2]
```diff
@@ -13,6 +13,15 @@ import (
 	"chainlink/core/utils"
 
 	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var (
+	headsReceived = promauto.NewCounter(prometheus.CounterOpts{
```
Comment (spooktheducks) [inline core/services/head_tracker.go]: Kind of a quibble, but using a prefix like `num` or `metric` might make this a little clearer to someone less familiar with this part of the codebase.

File: core/services/job_subscriber.go [hunk 3]
```diff
@@ -8,9 +8,18 @@ import (
 	"chainlink/core/store"
 	"chainlink/core/store/models"
 
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
 	"go.uber.org/multierr"
 )
 
+var (
+	jobSubscriptions = promauto.NewGauge(prometheus.GaugeOpts{
```
Comment (dimroc) [inline core/services/job_subscriber.go]: Optional: This one feels pretty left out of the `numbers` prefix party. `number` seems to only prefix counters, while gauges don't have one. Should we just have a prefix for anything prometheus? I like being able to know at a glance that it's metrics related. `jobSubscriptions` probably collides with a local variable somewhere in our code base.

`promJobSubscriptions`
`gaugeJobSubscriptions`
`metricsJobSubscriptions`

**Pull request merged**